muttlib.c

```diff
@@ -995,8 +995,9 @@ void mutt_expando_format(char *buf, size_t buflen, size_t col, int cols, const c
         /* eat the format string */
         cp = prefix;
         count = 0;
-        while ((count < (sizeof(prefix) - 1)) && (isdigit((unsigned char) *src) || (*src == '.') ||
-                                            (*src == '-') || (*src == '=')))
+        while ((count < (sizeof(prefix) - 1)) && (*src != '\0') &&
+               (isdigit((unsigned char) *src) || (*src == '.') ||
+                (*src == '-') || (*src == '=')))
```

Comment:
Technically, the check for (*src != '\0') is redundant here, because you are guaranteed that is the case if (isdigit || '.' || '-' || '=') succeeds.  But for symmetry with line 984, where the check is NOT redundant, I like keeping it in both places.
LGTM